datasets.py

```diff
@@ -62,7 +63,10 @@ def __getitem__(self, index):
 			ref_normalized, targets = zip(*targets)
 		else:
 			signal, sample_rate = audio.read_audio(audio_path, sample_rate = self.sample_rate, mono = self.mono, normalize = False, backend = self.audio_backend, duration=self.max_duration)
-			replace_transcript = not transcript or any(t.get('begin') is None and t.get('end') is None for t in transcript) and all(t.get('ref') is not None for t in transcript)
+			replace_transcript = self.join_transcript or \
+				not transcript or \
+				any(t.get('begin') is None and t.get('end') is None for t in transcript) and \
```

Comment:
please put `any(...) and all(...)` in round brackets. i think the semantics is already like this, but for clarity it'd be better